src/core/notifier/handler/notification/chart_handler.go

```diff
@@ -0,0 +1,107 @@
+package notification
+
+import (
+	"errors"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/core/notifier/model"
+	"github.com/goharbor/harbor/src/pkg/notification"
+)
+
+// ChartPreprocessHandler preprocess chart event data
+type ChartPreprocessHandler struct {
+}
+
+// Handle preprocess chart event data and then publish hook event
+func (cph *ChartPreprocessHandler) Handle(value interface{}) error {
+	// if global notification configured disabled, return directly
+	if !config.NotificationEnable() {
```

Comment:
if !config.WithChartMuseum() {
  return nil
}

Reply:
i think it has been judged when it gets here.